### PR TITLE
WIP: staticpods: make installer controller avoid down nodes after a day

### PR DESCRIPTION
**Pull request by sttts**
Addressing stuck, expired certificates after 30d of one master's downtime.

### CODE REVIEW

File: pkg/operator/staticpod/controller/installer/installer_controller.go [hunk 1]
```diff
@@ -286,36 +292,58 @@ type staticPodStateFunc func(ctx context.Context, nodeName string) (state static
 // - not updating
 // - ready
 // - at the revision claimed in CurrentRevision.
-func nodeToStartRevisionWith(ctx context.Context, getStaticPodStateFn staticPodStateFunc, nodes []operatorv1.NodeStatus) (int, string, error) {
-	if len(nodes) == 0 {
-		return 0, "", fmt.Errorf("nodes array cannot be empty")
+func nodeToStartRevisionWith(ctx context.Context, getStaticPodStateFn staticPodStateFunc, nodeStatuses []operatorv1.NodeStatus, nodes []*corev1.Node) (int, string, error) {
```
Comment (p0lyn0mial) [inline pkg/operator/staticpod/controller/installer/installer_controller.go:295]: could we document the order in which the nodes will be processed?

File: pkg/operator/staticpod/controller/installer/installer_controller.go [hunk 2]
```diff
@@ -286,36 +292,58 @@ type staticPodStateFunc func(ctx context.Context, nodeName string) (state static
...
+		}
+		nodeStatusesUpFirst = append(nodeStatusesUpFirst, status)
 	}
 
-	// find upgrading node as this will be the first to start new revision (to minimize number of down nodes)
-	for i := range nodes {
-		if nodes[i].TargetRevision != 0 {
-			reason := fmt.Sprintf("node %s is progressing towards %d", nodes[i].NodeName, nodes[i].TargetRevision)
+	// find upgrading node as this will be the first to start new revision (to minimize number of down nodeStatuses)
+	for i := range nodeStatuses {
```
Comment (p0lyn0mial) [inline pkg/operator/staticpod/controller/installer/installer_controller.go:323]: shouldn't you use `nodeStatusesUpFirst` ?

**Comment by p0lyn0mial** [comment]
We should also understand where did we fail in case one node was down. 
Was it here https://github.com/openshift/library-go/blob/master/pkg/operator/staticpod/controller/installer/installer_controller.go#L351 ?

File: pkg/operator/staticpod/controller/installer/installer_controller.go (hunk 1 above)
Comment (sttts) [reply pkg/operator/staticpod/controller/installer/installer_controller.go:295]: done

**Pull request closed without merging**